src/common/utils/history/index.ts

```diff
@@ -110,9 +110,16 @@ export const createLegacyHistory = (): BrowserHistory => {
     },
     ...(hasWindow()
       ? {
-          push: (path: string) => window.location.assign(path),
-          replace: (path: string) =>
-            window.history.replaceState({}, document.title, path),
+          push: (path: string | Location) =>
+            window.location.assign(
+              typeof path === 'string' ? path : createPath(path || {})
```

Comment:
Regarding null the behaviour change is here; null is not string so this is a behaviour change (but not an exception)